Simplify a function declaration.

bangerth opened the pull request:
I was looking for a simple enough example to show `requires` clauses in the release paper. This one wasn't quite it, but it wasn't hard to make it simple enough -- and there really wasn't a good reason for using such a complicated way to declare the function to begin with.

kronbichler commented:
There are some failing tests, which all look similar:
```
/jenkins/workspace/dealii-serial_PR-15255/tests/aniso/anisotropic_crash.cc:88: error: undefined reference to 'std::vector<dealii::Triangulation<2, 2>::active_cell_iterator, std::allocator<dealii::Triangulation<2, 2>::active_cell_iterator> > dealii::GridTools::find_cells_adjacent_to_vertex<dealii::Triangulation<2, 2> >(dealii::Triangulation<2, 2> const&, unsigned int)'
collect2: error: ld returned 1 exit status
```

bangerth commented:
Well, this sure was fascinating to track down. It turns out that we have the function's implementation in two `.cc` files. I changed the signature of the function in the `.h` file as well as one of the `.cc` files, but it was the old definition in the *other* `.cc` file that was explicitly instantiated. So we did instantiate the old signature, but the test saw the new signature in the `.h` file and consequently we got a linker error. Good stuff.

Fixed by removing the duplicate implementation in the `.cc` file where the function was not actually instantiated.

kronbichler commented:
Thank you.
